### PR TITLE
feat: Add Deno.ServeDefaultExport type

### PR DESCRIPTION
**Pull request by bartlomieju**
Closes https://github.com/denoland/deno/issues/23725

### CODE REVIEW

File: cli/tsc/dts/lib.deno.ns.d.ts [hunk 1]
```diff
@@ -6256,6 +6256,12 @@ declare namespace Deno {
     info: ServeHandlerInfo,
   ) => Response | Promise<Response>;
 
+  export interface Serve {
+    fetch: (
+      request: Request,
+    ) => Response | Promise<Response>;
+  }
```
Comment (ry) [inline cli/tsc/dts/lib.deno.ns.d.ts:6263]: Seems confusing that `Deno.Serve` and `Deno.serve` are unrelated.

Comment (bartlomieju) [reply cli/tsc/dts/lib.deno.ns.d.ts:6263]: Happy to hear suggestions for other names

Comment (ry) [reply cli/tsc/dts/lib.deno.ns.d.ts:6263]: ServeDefaultExport

**Review by marvinhagemeister — APPROVED** [review]
🚀

**Pull request merged**